pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/V2Operator.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.runtime.operator;
+
+import java.util.List;
+import org.apache.pinot.core.common.Operator;
+import org.apache.pinot.core.operator.BaseOperator;
+import org.apache.pinot.query.runtime.blocks.TransferableBlock;
+import org.slf4j.LoggerFactory;
+
+
+public abstract class V2Operator extends BaseOperator<TransferableBlock> implements AutoCloseable {
+  private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(MailboxReceiveOperator.class);
+
+  @Override
+  public List<Operator> getChildOperators() {
+    throw new UnsupportedOperationException();
+  }
+
+  abstract List<V2Operator> getV2ChildOperators();
+
+  @Override
+  public void close() {
```

Comment:
Could some operators take a while to close or do we expect all of them to close immediately (excluding the dfs cost)?

Reply:
yeah. that's a good point. I haven't figured out a good way to deal with this. Ideally each op close() call should return within request deadline or something. but the Autocloseable doesn't come with a deadline. 
 I put a TODO. I'll update the API later if needed.

Reply:
echo back to my shutdown comment. should we put a timeout in case shutdown is stuck?

Reply:
added a TODO